service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/CreateAndDeleteAzureControlledResourceFlightTest.java

```diff
@@ -203,27 +187,66 @@ public void createAndDeleteAzureRelayNamespaceControlledResource() throws Interr
   }
 
   @Test
-  public void createAzureDiskControlledResource() throws InterruptedException {
+  public void createAndDeleteAzureStorageResource() throws InterruptedException {
     UUID workspaceUuid = azureTestUtils.createWorkspace(workspaceService);
     AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    createCloudContext(workspaceUuid, userRequest);
 
-    // Cloud context needs to be created first
-    FlightState createAzureContextFlightState =
-        StairwayTestUtils.blockUntilFlightCompletes(
-            jobService.getStairway(),
-            CreateAzureContextFlight.class,
-            azureTestUtils.createAzureContextInputParameters(workspaceUuid, userRequest),
-            STAIRWAY_FLIGHT_TIMEOUT,
-            null);
+    final ApiAzureStorageCreationParameters creationParameters =
+            ControlledResourceFixtures.getAzureStorageCreationParameters();
 
-    assertEquals(FlightStatus.SUCCESS, createAzureContextFlightState.getFlightStatus());
-    assertTrue(
-        workspaceService.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isPresent());
+    final UUID resourceId = UUID.randomUUID();
+    ControlledAzureStorageResource resource =
+            ControlledAzureStorageResource.builder()
+                    .common(
+                            ControlledResourceFields.builder()
+                                    .workspaceUuid(workspaceUuid)
+                                    .resourceId(resourceId)
+                                    .name(getAzureName("rs"))
+                                    .description(getAzureName("rs-desc"))
+                                    .cloningInstructions(CloningInstructions.COPY_NOTHING)
+                                    .accessScope(AccessScopeType.fromApi(ApiAccessScope.SHARED_ACCESS))
+                                    .managedBy(ManagedByType.fromApi(ApiManagedBy.USER))
+                                    .build())
+                    .storageAccountName(creationParameters.getName())
+                    .region(creationParameters.getRegion())
+                    .build();
+
+    // Submit a storage account creation flight and then verify the resource exists in the workspace.
+    createResource(workspaceUuid, userRequest, resource, WsmResourceType.CONTROLLED_AZURE_STORAGE_ACCOUNT);
+
+    // Submit a storage account deletion flight.
+    FlightState deleteFlightState =
+            StairwayTestUtils.blockUntilFlightCompletes(
+                    jobService.getStairway(),
+                    DeleteControlledResourceFlight.class,
+                    azureTestUtils.deleteControlledResourceInputParameters(
+                            workspaceUuid, resourceId, userRequest, resource),
+                    STAIRWAY_FLIGHT_TIMEOUT,
+                    null);
+    assertEquals(FlightStatus.SUCCESS, deleteFlightState.getFlightStatus());
+
+    TimeUnit.SECONDS.sleep(5);
```

Comment:
Is this sleep required for the resource deletion to fully propagate through Azure?

Reply:
It's what was in the relay test that I used as the basis for this new test… I do generally hate sleeps and could be convinced to rework it to poll.